core/templates/pages/admin-page/misc-tab/admin-misc-tab.component.ts

```diff
@@ -60,13 +76,138 @@ export class AdminMiscTabComponent {
   message: string = '';
   expIdToGetInteractionIdsFor!: string;
   explorationInteractionIds: string[] = [];
+  directiveSubscriptions = new Subscription();
+  loadingMessage: string = '';
+  selectedUserGroup: string = '';
+  newUserGroupName: string = '';
+  removable: boolean = true;
+  selectable: boolean = true;
+  separatorKeysCodes: number[] = [ENTER];
+  userGroupToUsersMapBackup: Record<string, string[]> = {};
+  userGroupsToUsers: Record<string, string[]> = {};
 
   constructor(
     private adminBackendApiService: AdminBackendApiService,
+    private adminDataService: AdminDataService,
     private adminTaskManagerService: AdminTaskManagerService,
-    private windowRef: WindowRef
+    private alertsService: AlertsService,
+    private windowRef: WindowRef,
+    private loaderService: LoaderService
   ) {}
 
+  async fetchUserGroupData(): Promise<void> {
+    const data = await this.adminDataService.getDataAsync();
+    this.userGroupsToUsers = data.userGroups;
+    this.userGroupToUsersMapBackup = cloneDeep(this.userGroupsToUsers);
+    this.loaderService.hideLoadingScreen();
+  }
+
+  resetUserGroups(): void {
+    if (this.areUserGroupsUpdated()) {
+      if (
+        !this.windowRef.nativeWindow.confirm(
+          'This will revert all changes you made. Are you sure?'
+        )
+      ) {
+        return;
+      }
+      this.userGroupsToUsers = cloneDeep(this.userGroupToUsersMapBackup);
+    }
+  }
+
+  areUserGroupsUpdated(): boolean {
+    return !isEqual(this.userGroupsToUsers, this.userGroupToUsersMapBackup);
+  }
+
+  getUserGroups(): string[] {
+    let userGroups: string[] = [];
+    for (let key in this.userGroupsToUsers) {
+      userGroups.push(key);
+    }
+    return userGroups;
+  }
+
+  addUser(event: {value: string}): void {
+    const value = (event.value || '').trim();
+    if (!value || value === '') {
+      return;
+    }
+
+    if (this.userGroupsToUsers[this.selectedUserGroup].includes(value)) {
+      this.alertsService.addWarning(
+        `The user ${value} already exists in the user ` +
+          `group ${this.selectedUserGroup}.`
+      );
+      return;
+    }
+    this.userGroupsToUsers[this.selectedUserGroup].push(value);
+    this.userInput.nativeElement.value = '';
+  }
+
+  removeUser(username: string): void {
```

Comment:
Rename method name to `removeUserFromSelectedGroup`.

Reply:
Updated